.github/workflows/CD.yml

```
@@ -634,7 +638,7 @@ jobs:
           aws-region: us-west-2
 
       - name: Mirror ADOT operator
-        if: ${{ steps.release-latest-image.outputs.cache-hit != 'true' && (steps.version.outputs.major-update == 'true' || steps.version.outputs.minor-update == 'true') || steps.version.outputs.same-version == 'true' }}
+        if: ${{ github.event.inputs.skip-adot-operator-mirroring == 'false' && (steps.release-latest-image.outputs.cache-hit != 'true' && (steps.version.outputs.major-update == 'true' || steps.version.outputs.minor-update == 'true') || steps.version.outputs.same-version == 'true') }}
```

```suggestion
        if: ${{ github.event.inputs.skip-adot-operator-mirroring == 'false' && (steps.release-latest-image.outputs.cache-hit != 'true' && (steps.version.outputs.patch-update == 'false') }}
```

Can you verify that this logic is the same by referencing [this](https://github.com/aws-observability/aws-otel-collector/blob/main/tools/workflow/docker-version-compare.sh) script.

---

Not sure why it checks for `steps.version.outputs.same-version == 'true'` so that should be removed; checking only `steps.version.outputs.patch-update == 'false'` doesn't account for if major/minor updates have happened. If we check it as `steps.version.outputs.patch-update == 'true'` though it should be equivalent, while also adding a check for patch updates (if we want that in adot operator images)

---

I believe only `xxx-update` values can be true at once. I believe `any-update` can also be true but that is not referenced here.